roiextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -429,124 +429,123 @@ def get_property_data(self, property_name):
         return ret_val
 
     @staticmethod
-    def write_segmentation(segext_obj, savepath, metadata_dict, **kwargs):
+    def write_segmentation(segext_obj, savepath, metadata_dict=None, **kwargs):
+        source_path = segext_obj.filepath
+        print(f'writing nwb for {segext_obj.extractor_name}\n')
         if isinstance(metadata_dict, str):
             with open(metadata_dict, 'r') as f:
-                metadata_dict = yaml.safe_load(f)
+                metadata = yaml.safe_load(f)
 
+        metadata_dict = metadata
         # NWBfile:
         nwbfile_args = dict(identifier=str(uuid.uuid4()), )
-        if 'NWBFile' in metadata_dict:
-            nwbfile_args.update(**metadata_dict['NWBFile'])
+        nwbfile_args.update(**metadata_dict['NWBFile'])
         nwbfile = NWBFile(**nwbfile_args)
 
         # Subject:
-        if 'Subject' in metadata_dict:
-            nwbfile.subject = Subject(**metadata_dict['Subject'])
+        nwbfile.subject = Subject(**metadata_dict['Subject'])
 
         # Device:
-        if isinstance(metadata_dict['ophys']['Device'], list):
-            for devices in metadata_dict['ophys']['Device']:
+        if isinstance(metadata_dict['Ophys']['Device'], list):
+            for devices in metadata_dict['Ophys']['Device']:
                 nwbfile.create_device(**devices)
         else:
-            nwbfile.create_device(**metadata_dict['ophys']['Device'])
+            nwbfile.create_device(**metadata_dict['Ophys']['Device'])
 
         # Processing Module:
-        ophys_mod = nwbfile.create_processing_module('ophys',
+        ophys_mod = nwbfile.create_processing_module('Ophys',
```

Comment:
"ophys" with a lower case "o" is correct. See https://www.nwb.org/best-practices/. I know there are other files and conversion scripts out there that use a capital "O." Those are mistakes that need to be corrected.